src/ctia/import/threatgrid/feed/sightings.clj

```diff
@@ -7,76 +7,53 @@
             [cheshire.core :as json]
             [clojure.edn :as edn]))
 
-(defn make-feed-sighting
-  [description feed-uri timestamp observable indicator]
-  (let [sighting {:description description
-                  :type "sighting"
-                  :source "ThreatGrid Feed"
-                  :source_uri feed-uri
-                  :tlp "green"
-                  :confidence "High"
-                  :timestamp timestamp
-                  :observables [observable]
-                  :indicators [indicator]}]))
+(defn entry->relations
+  [feed entry source-observable relation-type related-type related-key]
+  (mapv (fn [related-value]
+          {:origin (:source feed)
+           :origin_uri (:source_uri feed)
+           :source source-observable
+           :relation relation-type
+           :related {:value related-value
+                     :type related-type}})
+        (related-key entry)))
 
 (defn transform
-  [entries
-   observable-type
-   observable-value
-   indicator-id
-   indicator-source
-   source-uri & {:keys [tlp confidence]
-                 :or {tlp "green"
-                      confidence "High"}}]
-  (map
-   (fn [entry]
-     (let [{:keys [description timestamp]} entry]
-       {:description description
-        :type "sighting"
-        :source "ThreatGrid"
-        :source_uri source-uri
-        :tlp tlp
-        :confidence confidence
-        :timestamp timestamp
-        :observables [{:type observable-type
-                       :value (observable-value entry)}]
-        :indicators [{:confidence "High"
-                      :source indicator-source
-                      :indicator_id indicator-id}]}))
-   entries))
-
-(defn entries->sightings
-  [entries observable-type observable-field
-   indicator-id indicator-source source-uri & {:as options}]
-  (apply transform entries observable-type observable-field
-         indicator-id indicator-source source-uri options))
+  [feed & {:keys [tlp confidence]
+           :or {tlp "green"
+                confidence "High"}}]
+  (let [{:keys [title description source source_uri
```

Comment:
@saintx in case you didn't know this is valid Clojure:

```clj
{title :title
 description :description
 indicator-source :source
 source-uri :source_uri
 observable :observable
 indicator :indicator
 entries :entries} feed
```

This means you wouldn't need to have the `let` lines for `indicator-source source` and `source-uri source_uri`

You can actually do this, and you wouldn't need the `let` at all:

```clj
(defn transform 
  [{title :title
            description :description
            indicator-source :source
            source-uri :source_uri
            {observable-type :type
             observable-field :field
             :as observable} :observable
            {indicator-id :id
             :as indicator} :indicator
            entries :entries
            :as feed}
 & {:keys [tlp confidence]
           :or {tlp "green"
                confidence "High"}}]

...
)
```